packages/app-mobile/components/screens/NoteRevisionViewer.tsx

```diff
@@ -142,6 +145,51 @@ const NoteRevisionViewer: React.FC<Props> = props => {
 		}
 	}, [note]);
 
+	const [deleting, setDeleting] = useState(false);
+	const deleteAllButton_onPress = async () => {
+		if (!noteId) return;
+		const response = await shim.showMessageBox(RevisionService.instance().deleteAllPromptMessage(), {
+			title: _('Warning'),
+			buttons: [_('Yes'), _('No')],
+			type: MessageBoxType.Confirm,
+		});
+		if (response === 0) {
+			setDeleting(true);
+			try {
+				await Revision.deleteAllRevisionsForNote(noteId);
+				await shim.showMessageBox(RevisionService.instance().deleteAllSuccessMessage(), { type: MessageBoxType.Info });
+			} finally {
+				setDeleting(false);
+				setCurrentRevisionId(null);
+				setHasRevisions(false);
+				revisions.length = 0;
+				options.length = 0;
+			}
+		}
+	};
+
+	let menuOptionsCache_: Record<string, MenuOptionType[]> = {};
```

Comment:
At present, `menuOptionsCache_` is recreated on each render (and thus the menu options aren't cached).

Consider either:
- Using one instance of `menuOptionsCache_` for each component (e.g. with a `useRef`):
  ```suggestion
	  let menuOptionsCache_ = useRef<Record<string, MenuOptionType[]>>({}).current;
  ```

- (Recommended) Creating the menu options using a `useMemo`. For this option, `const menuOptions = ...` might be replaced with something similar to:
  ```tsx
  const disableDeleteAll = deleting || !hasRevisions;
  const menuOptions = useMemo(() => {
  	const output: MenuOptionType[] = [{
  		title: _('Delete All'),
  		onPress: async () => {
  			void deleteAllButton_onPress();
  		},
  		disabled: disableDeleteAll,
  	}];

  	return output;
  }, [disableDeleteAll]);
  ```